File: 2022/09/25186.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n");
const N = BigInt(input[0]);
const D = input[1].split(" ").map(BigInt);
const sum = D.reduce((a, b) => BigInt(a) + BigInt(b));

function solution() {
    if (N === 1 && D[0] === 1) return "Happy";
    for (let i = 0; i < N; i++) {
        if (sum - D[i] < D[i]) return "Unhappy";
    }
    return "Happy";
}
console.log(solution());

File: 2022/09/25186.py
input = __import__('sys').stdin.readline
N = int(input())
D = list(map(int, input().split()))
total = sum(D)


def solution():
    if(N == 1 and D[0] == 1):
        return 'Happy'
    for i in D:
        if total - i < i:
            return 'Unhappy'
    return 'Happy'


print(solution())
